implementations/rust/ockam/ockam_command/src/tcp/listener/show.rs

```diff
@@ -0,0 +1,57 @@
+use clap::Args;
+
+use crate::node::default_node_name;
+use crate::util::extract_address_value;
+use ockam::Context;
+use ockam_api::nodes::models;
+use ockam_core::api::Request;
+
+use crate::util::{node_rpc, Rpc};
+use crate::CommandGlobalOpts;
+
+#[derive(Clone, Debug, Args)]
+pub struct ShowCommand {
+    #[command(flatten)]
+    node_opts: TCPlistenerNodeOpts,
+
+    /// Tcp Listener ID
+    pub id: String,
+}
+
+#[derive(Clone, Debug, Args)]
+pub struct TCPlistenerNodeOpts {
+    /// Node at which to create the listener
+    #[arg(global = true, long, value_name = "NODE", default_value_t = default_node_name())]
+    pub at: String,
+}
+
+impl ShowCommand {
+    pub fn run(self, options: CommandGlobalOpts) {
+        node_rpc(run_impl, (options, self));
+    }
+}
+
+async fn run_impl(
+    ctx: Context,
+    (opts, cmd): (CommandGlobalOpts, ShowCommand),
+) -> crate::Result<()> {
+    let node = extract_address_value(&cmd.node_opts.at)?;
+    let mut rpc = Rpc::background(&ctx, &opts, &node)?;
+    rpc.request(Request::get("/node/tcp/listener")).await?;
```

Comment:
You have to add a new endpoint to the NodeManager to respond to this new request. Note that the endpoint must include the `id` of the tcp-listener you want to retrieve.

Check this PR as an example: https://github.com/build-trust/ockam/pull/4446